plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StringVersionComparator.java

```diff
@@ -0,0 +1,23 @@
+
+package com.google.cloud.tools.eclipse.appengine.facets;
+
+import com.google.common.collect.Ordering;
+import java.util.Comparator;
+
+/**
+ * A {@link Comparator<String>} for the Faceted Project framework that compares versions expressed
+ * as strings.
+ */
+public class StringVersionComparator implements Comparator<String> {
+
+  // Since this comparator is usually specified via plugin.xml, one would normally
+  // implement this as an IExecutableExtensionFactory that returns Ordering.natural().
+  // But the fproj framework instantiates the specified class *directly* and so we
+  // must actually implement the comparator.
+
+  @Override
+  public int compare(String o1, String o2) {
+    return Ordering.natural().compare(o1, o2);
```

Comment:
I'll document that this is for pure-string comparisons, like "JRE7" vs "JRE8"; a version comparator isn't necessary for numeric versions as the faceted project framework handles that out of the box.